packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerHighlight.tsx

```diff
@@ -127,4 +162,33 @@ const useHighlighter = ({
   }, [document, isReady, pageNum, pdfRenderedText, processedDoc, textMappings]);
 };
 
+function useScrollIntoActiveHighlight(
+  highlightDivRef: React.MutableRefObject<HTMLDivElement | null>,
+  shapes: (HighlightShape | undefined)[],
+  activeIds: string[] | undefined
+) {
+  useEffect(() => {
+    if (!highlightDivRef.current) {
+      return;
+    }
+
+    const activeShape = shapes.find(
+      shape => shape?.highlightId && activeIds?.includes(shape.highlightId)
+    );
+    if (activeShape) {
+      const timer = setTimeout(() => {
+        const highlightDiv = highlightDivRef.current;
+        if (!highlightDiv) return;
+
+        const highlightElm = highlightDiv?.querySelector(
+          `[data-highlight-id=${activeShape.highlightId}]`
+        );
+        highlightElm?.firstElementChild?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+      }, 50);
+      return () => clearTimeout(timer);
```

Comment:
If I'm reading this right, this is meant to be the cleanup function for the `useEffect`, right? If so, please comment.

Or better yet, do it like they show in the [docs](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1):

```
return function cleanup() {
  clearTimeout(timer);
}
```